@here/harp-mapview/lib/geometry/SolidLineMesh.ts

```diff
@@ -0,0 +1,451 @@
+): void {
+    const vExt = tmpV1;
+    const plane = tmpPlane;
+    const interPlane = tmpV2;
+    const line = tmpLine1;
+
+    const geometry = mesh.geometry as THREE.BufferGeometry;
+    const attributes = geometry.attributes;
+    const position = attributes.position as THREE.BufferAttribute;
+    const bitangent = attributes.bitangent;
+    const indices = geometry.index!.array;
+
+    tmpSphere.copy(bSphere);
+    tmpSphere.applyMatrix4(mesh.matrixWorld);
+    tmpSphere.radius += halfWidth;
+
+    if (!raycaster.ray.intersectsSphere(tmpSphere)) {
+        return;
+    }
+
+    for (let i = beginIdx; i < endIdx; i += SEGMENT_STRIDE) {
+        const a = indices[i];
+        const b = indices[i + VERTEX_STRIDE];
+
+        // Find the plane containing the line segment, using the segment start, end and extrusion
+        // vector.
+        line.start.fromBufferAttribute(position, a);
+        line.end.fromBufferAttribute(position, b);
+        vExt.set(bitangent.getX(a), bitangent.getY(a), bitangent.getZ(a)).normalize();
+        plane.setFromCoplanarPoints(line.start, tmpV3.copy(line.start).add(vExt), line.end);
+
+        // The ray intersection if any, will be on the extrusion plane.
+        if (!localRay.intersectPlane(plane, interPlane)) {
+            continue;
+        }
+
+        const interLine = tmpV3;
+        if (
+            !intersectLine(
+                localRay,
+                line,
+                vExt,
+                lHalfWidth,
+                lHalfWidthSq,
+                plane,
+                interPlane,
+                interLine
+            )
+        ) {
+            continue;
+        }
+
+        // Move back to world space for distance calculation
+        const interLineWorld = interLine.applyMatrix4(mesh.matrixWorld);
+
+        const distance = raycaster.ray.origin.distanceTo(interLineWorld);
+
+        if (distance < raycaster.near || distance > raycaster.far) {
+            continue;
+        }
+
+        intersects.push({
+            distance,
+            point: interLineWorld.clone(),
+            index: i,
+            object: mesh
+        });
+    }
+}
+
+const singleFeatureStart = [0];
+
+/**
+ * Finds the intersections of a ray with a group within a solid line mesh.
+ * @param mesh The mesh whose intersections will be found.
+ * @param material The material used by the group inside the mesh.
+ * @param raycaster  Contains the intersection ray.
+ * @param localRay Same ray as raycaster.ray but in object local space.
+ * @param intersects  Array that will contain all intersections found between ray and feature.
+ * @param firstFeatureIdx Index of the first feature in the group.
+ * @param groupEndIdx Index of the last vertex in the group.
+ * @returns The next feature index after the group.
+ */
+function intersectGroup(
+    mesh: THREE.Mesh,
+    material: THREE.Material,
+    raycaster: THREE.Raycaster,
+    localRay: THREE.Ray,
+    intersects: THREE.Intersection[],
```

Comment:
Same as above... output should be at the end and described (mostly done) as output.

Reply:
done.